src/components/_app/useKeyPairContext.tsx

```diff
@@ -180,7 +180,278 @@ const setKeyPair = async ({
 
 const checkKeyPair = ([_, savedPubKey, pubKey]): boolean => savedPubKey === pubKey
 
-const useKeyPair = () => {
+const KeyPairContext = createContext<{
+  ready: boolean
+  pubKey: string | undefined
+  keyPair: CryptoKeyPair | undefined
+  isValid: boolean
+  set: any
+}>({
+  ready: false,
+  pubKey: undefined,
+  keyPair: undefined,
+  isValid: false,
+  set: () => {},
+})
+
+const KeyPairProvider = ({ children }: PropsWithChildren<unknown>): JSX.Element => {
```

Comment:
The file name could be `KeyPairProvider` so we stay consistent ;) *(at least under this directory, haha)*

Reply:
The other files ended with the name Provider export only the Context and the Provider itself, but this one exports some other tools. What do you think about moving the IDB-related util function into a different file in a global util dictionary?

Reply:
Good point, but since both `getKeyPairFromIdb` and `setKeyPairToIdb` are keypair-related functions, I think we can leave them in this file and just export them from here ;)